src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,102 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int GROWTH_COEFFICIENT = 2;
+    private int capacity = 16;
+    private int threshold = (int) (capacity * LOAD_FACTOR);
+    private int size;
+    private Node<K, V>[] table = new Node[capacity];
+
+    private static class Node<K, T> {
+        private K key;
+        private T value;
+        private Node<K, T> next;
+
+        public Node(K key, T value, Node<K, T> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    private int findIndex(K key) {
+        if (key == null) {
+            return 0;
+        } else if (key.hashCode() % capacity >= 0) {
+            return key.hashCode() % capacity;
+        } else {
+            return -key.hashCode() % capacity;
+        }
+    }
+
+    private void resize() {
+        threshold = (int) (threshold * GROWTH_COEFFICIENT);
+        size = 0;
+        Node<K, V>[] oldTable = table;
+        table = new Node[oldTable.length * GROWTH_COEFFICIENT];
+        for (Node<K, V> node : oldTable) {
+            while (node != null) {
+                put(node.key, node.value);
+                node = node.next;
+            }
+        }
+    }
 
     @Override
     public void put(K key, V value) {
+        if (threshold == size) {
+            resize();
+        }
 
+        int index = findIndex(key);
+        Node<K, V> newNode = new Node<>(key, value, null);
+        if (table[index] == null) {
+            table[index] = newNode;
+        } else {
+            Node<K, V> currentNode = table[index];
+            Node previousNode = null;
+            while (currentNode != null) {
+                if (key == null) {
+                    if (key == currentNode.key) {
+                        currentNode.value = value;
+                        return;
+                    }
+                } else {
+                    if (key.equals(currentNode.key)) {
+                        currentNode.value = value;
+                        return;
+                    }
```

Comment:
could we combine these if's?

Reply:
great note, thank you! It looked bizzare to me, but I didn't know how to improve. Now the code is much prettier)